lib/modules/coverage/contract_source.js

```diff
@@ -0,0 +1,306 @@
+        case 'SourceUnit':
+          children = node.nodes;
+          break;
+
+        case 'PragmaDirective':
+          // These nodes do nothing, so we move on.
+          break;
+
+        case 'FunctionDefinition':
+          // Istanbul only wants the function definition, not the body, so we're
+          // going to do some fun math here.
+          var functionSourceMap = new SourceMap(node.src);
+          var functionParametersSourceMap = new SourceMap(node.parameters.src);
+
+          var functionDefinitionSourceMap = new SourceMap(
+            functionSourceMap.offset,
+            (functionParametersSourceMap.offset + functionParametersSourceMap.length) - functionSourceMap.offset
+          ).toString();
+
+          var fnName = node.isConstructor ? "(constructor)" : node.name;
+          location = this.sourceMapToLocations(functionDefinitionSourceMap);
+
+          coverage.f[node.id] = 0;
+          coverage.fnMap[node.id] = {
+            name: fnName,
+            line: location.start.line,
+            loc: location
+          };
+
+          // Record function positions.
+          sourceMapToNodeType[node.src] = [{type: 'f', id: node.id, body: coverage.fnMap[node.id]}];
+
+          if(node.body) children = node.body.statements;
+          markLocations = [location];
+          break;
+
+        default:
+          // console.log(`Don't know how to handle node type ${node.nodeType}`);
+          break;
+      }
+
+      nodesRequiringVisiting = nodesRequiringVisiting.concat(children);
+
+      markLocations.forEach((location) => {
+        for(var i = location.start.line; i <= location.end.line; i++) {
+          coverage.l[i] = 0;
+        }
+      });
+
+    } while(nodesRequiringVisiting.length > 0);
+
+    var contractMatches = true;
+    for(var contractName in this.contractBytecode) {
+      var bytecode = this.contractBytecode[contractName];
+
+      // Try to match the contract to the bytecode. If it doesn't,
+      // then we bail.
+      trace.structLogs.forEach((step) => {
+        if(!contractMatches) return;
+
+        if(!bytecode[step.pc]) {
+          // This contract won't match so we bail
+          contractMatches = false;
+          return;
+        }
+      });
+
+      if(!contractMatches) break;
+
+      trace.structLogs.forEach((step) => {
+        step = bytecode[step.pc];
+        if(!step.sourceMap || step.sourceMap == '') return;
+
+        var nodes = sourceMapToNodeType[step.sourceMap];
+
+        if(!nodes) return;
+
+        var recordedLineHit = false;
+
+        nodes.forEach((node) => {
+          if(node.body && node.body.loc && !recordedLineHit) {
+            for(var line = node.body.loc.start.line; line <= node.body.loc.end.line; line++) {
+              coverage.l[line]++;
+            }
+
+            recordedLineHit = true;
+          }
+
+          if(node.type != 'b') coverage[node.type][node.id]++;
+
+          if(!node.parent) return;
+
+          switch(node.parent.type) {
```

Comment:
This switch is only for one condition. Can you convert to an `if` or is there supposed to be more conditions added later?

Reply:
Whenever we add Vyper support, there will be more conditions here.